### PR TITLE
Two tests for variants

### PR DESCRIPTION
**Pull request by konnov**
Experimental PR that migrates two tests to the variant type. This works even better than I expected.

Also, deleted two old test files that are not used in the integration tests.

[unreleased]: https://github.com/informalsystems/apalache/tree/unstable/.unreleased

### CODE REVIEW

File: test/tla/Bug1682.tla [hunk 1]
```diff
@@ -1,20 +1,44 @@
 ---- MODULE Bug1682 ----
 
-EXTENDS Integers
+EXTENDS Integers, Variants
+
+(*
+ @typeAlias: MESSAGE =
+      RequestVote({ term: Int, from: Str })
+    | Vote({ from: Str, to: Str, term: Int })
+    | AppendEntries({ term: Int, from: Str }) ;
```
Comment (Kukovec) [inline test/tla/Bug1682.tla:9]: Not sure I like the `({ ... })` double delimiter syntax in practice.

File: test/tla/TwoPhaseTyped_typedefs.tla [hunk 2]
```diff
@@ -0,0 +1,7 @@
+----------------------- MODULE TwoPhaseTyped_typedefs ----------------
+(*
+ @typeAlias: MESSAGE = Commit(NIL) | Abort(NIL) | Prepared(RM);
```
Comment (Kukovec) [inline test/tla/TwoPhaseTyped_typedefs.tla:3]: This makes me wonder, should we reserve a `Unit`-like type for stuff like this, so it isn't ad-hoc implemented?

**Review by Kukovec — APPROVED** [review]

File: test/tla/Bug1682.tla (hunk 1 above)
Comment (konnov) [reply test/tla/Bug1682.tla:9]: You can have any type there, `{ ... }` is a record

File: test/tla/TwoPhaseTyped_typedefs.tla (hunk 2 above)
Comment (Kukovec) [reply test/tla/TwoPhaseTyped_typedefs.tla:3]: Yes, but what I'm saying is we should create a list of reserved words (not just for variants, globally), and `Unit` or `NIL` or something alike should be on it. You can't expect people to figure out this idiom from a GH issue.

Comment (Kukovec) [reply test/tla/TwoPhaseTyped_typedefs.tla:3]: Followed by static analysis for reserved words.

Comment (konnov) [reply test/tla/TwoPhaseTyped_typedefs.tla:3]: We will write about it in the howto and the tutorial.

**Pull request merged**